systems_cross_building/windows_subsystems.rst

```diff
@@ -0,0 +1,193 @@
+- You can use them together with any other compiler to build Windows-native software, even with Visual
+  Studio.
+- You can use them with MinGW to build specific software for the subsystem, with a dependency to a
+  runtime DLL (``msys-2.0.dll`` and ``cygwin1.dll``)
+
+
+If you are building specific software for the subsystem, you have to specify a value for the setting ``os.subsystem``,
+if you are only using the subsystem for taking benefit of the UNIX tools but generating native Windows software, you
+shouldn't specify it.
+
+
+Running commands inside the subsystem
+-------------------------------------
+
+self.run()
+__________
+
+In a Conan recipe, you can use the ``self.run`` method specifying the parameter ``win_bash=True``
+that will call automatically to the tool :ref:`tools.run_in_windows_bash<run_in_windows_bash_tool>`.
+
+It will use the **bash** in the path or the **bash** specified for the environment variable :ref:`CONAN_BASH_PATH<conan_bash_path_env>`
+to run the specified command.
+
+Conan will automatically escape the command to match the detected subsystem.
+If you also specify the ``msys_mingw`` parameter to False, and the subsystem is ``MSYS2`` it will
+run in Windows-native mode, the compiler won't link against the ``msys-2.0.dll``.
+
+
+AutoToolsBuildEnvironment
+__________________________
+
+In the constructor of the build helper, you have the ``win_bash`` parameter. Set it to ``True`` to
+run the ``configure`` and ``make`` commands inside a bash.
+
+
+Controlling the build environment
+---------------------------------
+
+Building software in a Windows subsystem for a different compiler than MinGW can be painful sometimes.
+The reason is how the subsystem finds your compiler/tools in your system.
+
+For example, the `icu <http://site.icu-project.org/>`_ library requires Visual Studio to be built in Windows, but also a subsystem
+able to build the Makefile. A very common problem and example of the pain is the ``link.exe`` program.
+In the Visual Studio suite, ``link.exe`` is the linker, but in the ``MSYS2`` environment the ``link.exe``
+is a tool to manage symbolic links.
+
+Conan is able to prioritize the tools when you use ``build_requires``, and put the tools in the PATH in
+the right order.
+
+There are some packages you can use as ``build_requires``:
+
+
+- From Bincrafters repository: ``conan remote add bincrafters https://api.bintray.com/conan/bincrafters/public-conan``
+
+    - **msys2_installer/latest@bincrafters/stable**: MSYS2 subsystem as a Conan package.
+    - **cygwin_installer/2.9.0@bincrafters/stable**: Cygwin subsystem as a Conan package.
+
+- From Conan-center:
+
+    - **mingw_installer/1.0@conan/stable**: MinGW compiler installer as a Conan package.
+
+- From Conan-community: ``conan remote add community https://api.bintray.com/conan/conan-community/conan``
+
+    - **vs_path_wrapper/1.0@conan/stable**: Wrapper of your own Visual Studio compiler, to be able to put it
```

Comment:
I still don't see why this deserves a build require, cannot be made with a ``tool.vs_path_order()`` helper? Cannot be coded in the recipe to be self-contained and do not require an external build-require?

Reply:
I will try to modify the current `tool.vcvars` context manager, but the problem is, the run_in_windows_bash function reads the `conanfile.env` to put these vars first and then all the system ones (os.environ) inside the bash subsystem, so it doesn't matter if I put first the visual in the `os.environ`. It will require some hack